src/can.c

```diff
@@ -120,6 +121,18 @@ static void thingset_can_addr_claim_tx_cb(const struct device *dev, int error, v
     }
 }
 
+static void thingset_can_addr_discover_tx_cb(const struct device *dev, int error, void *user_data)
+{
+    struct thingset_can *ts_can = user_data;
+
+    if (error == 0) {
+        k_event_post(&ts_can->events, EVENT_ADDRESS_DISCOVER_MSG_SENT);
```

Comment:
I'm not sure I understand correctly, but it looks like this new event is never waited for.